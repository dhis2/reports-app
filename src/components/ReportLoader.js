import { CircularProgress } from '@dhis2/d2-ui-core'
import isEmpty from 'lodash.isempty'
import PropTypes from 'prop-types'
import React from 'react'
import { reportContent } from '../utils/react/propTypes'

const ReportLoader = ({ isLoading, content, children }) => {
    if (isLoading) {
        return (
            <div className="report-loader">
                <CircularProgress />
                <style jsx>{`
                    div {
                        margin: 48px 0;
                        text-align: center;
                    }
                `}</style>
            </div>
        )
    }

    if (isEmpty(content)) {
        return null
    }

    return children
}

ReportLoader.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    content: reportContent.isRequired,
    children: PropTypes.node.isRequired,
}

export default ReportLoader
