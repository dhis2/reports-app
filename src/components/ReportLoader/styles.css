.report-loader {
    margin: 48px 0;
    text-align: center;
}
