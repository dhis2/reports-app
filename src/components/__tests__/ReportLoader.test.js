import React from 'react'
import { shallow } from 'enzyme'
import ReportLoader from '../ReportLoader'

describe('<ReportLoader/>', () => {
    it('Renders a loader when isLoading is true', () => {
        const wrapper = shallow(
            <ReportLoader content={{}} isLoading={true}>
                <div>Hello world</div>
            </ReportLoader>
        )
        expect(wrapper).toMatchSnapshot()
    })

    it('Renders nothing when content is an empty object and isLoading is false', () => {
        const wrapper = shallow(
            <ReportLoader content={{}} isLoading={false}>
                <div>Hello world</div>
            </ReportLoader>
        )
        expect(wrapper.html()).toBeNull()
    })

    it('Renders its children when isLoading is false and content is non-empty', () => {
        const props = {
            content: { text: 'hello world' },
            isLoading: false,
        }
        const wrapper = shallow(
            <ReportLoader {...props}>
                <div>Hello world</div>
            </ReportLoader>
        )
        expect(wrapper.html()).toMatchSnapshot()
    })
})
