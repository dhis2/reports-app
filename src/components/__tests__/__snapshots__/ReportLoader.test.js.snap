// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportLoader/> Renders a loader when isLoading is true 1`] = `
<div
  className="report-loader"
>
  <CircularProgress
    large={false}
    small={false}
    style={Object {}}
  />
  <style>
    
                        div {
                            margin: 48px 0;
                            text-align: center;
                        }
                    
  </style>
</div>
`;

exports[`<ReportLoader/> Renders its children when isLoading is false and content is non-empty 1`] = `"<div>Hello world</div>"`;
